src/App.tsx

```diff
@@ -1,61 +1,94 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const [currentTitle, setCurrentTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [todos, setTodos] = useState(todosFromServer);
+  const [wasSubmitted, setWasSubmitted] = useState(false);
+
+  const isFormValid = currentTitle.trim() && currentUserId;
+
+  const formReset = () => {
+    setCurrentTitle('');
+    setCurrentUserId(0);
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isFormValid) {
+      const newTodo = {
+        id: Math.max(...todos.map(todo => todo.id)) + 1,
+        title: currentTitle,
+        userId: currentUserId,
+        completed: false,
+        user: usersFromServer.find(user => user.id === currentUserId) || '',
```

Comment:
[REACT KNOWLEDGE] - Don't render the component if the property that you pass to the component has `null` or `undefined` value. Here, you're assigning an empty string to user if the user is not found. Instead, you should assign `null` or `undefined` and handle the absence of a user properly in the component that receives this object.